zerver/views/invite.py

```diff
@@ -74,12 +74,28 @@ def get_user_invites(request: HttpRequest, user_profile: UserProfile) -> HttpRes
 @has_request_variables
 def revoke_user_invite(request: HttpRequest, user_profile: UserProfile,
                        prereg_id: int) -> HttpResponse:
-    do_revoke_user_invite(prereg_id, user_profile.realm_id)
+    try:
+        prereg_user = PreregistrationUser.objects.get(id=prereg_id)
+    except PreregistrationUser.DoesNotExist:
+        raise JsonableError(_("Invitation does not exist"))
+
+    if prereg_user.referred_by.realm != user_profile.realm:
+        raise JsonableError(_("Invitation does not exist"))
```

Comment:
Ah, good thought -- this does feel more like the view's processing of the raw request parameters than it feels like the core library logic.

Reply:
Also good to fix this message. Maybe "No such invitation"? That feels a little clearer that this is talking about the specific invitation the requester tried to refer to.